crates/fj-math/src/transform.rs

```diff
@@ -89,13 +91,16 @@ impl Transform {
         fovy: f64,
         znear: f64,
         zfar: f64,
-    ) -> [f64; 16] {
+    ) -> [Scalar; 16] {
         let projection = Perspective3::new(aspect_ratio, fovy, znear, zfar);
-        let mut res = [0f64; 16];
-        res.copy_from_slice(
-            (projection.to_projective() * self.0).matrix().as_slice(),
-        );
-        res
+        (projection.to_projective() * self.0)
+            .matrix()
+            .as_slice()
+            .iter()
+            .map(|f| Scalar::from(*f))
+            .collect::<Vec<Scalar>>()
```

Comment:
It's a bit unfortunate that we're going through a `Vec` here, due to the heap allocation that's probably expensive. I don't know how to change it off the top of my head, however. I'd have to experiment with it hands-on.

Reply:
I also wanted to avoid a Vec here. You can't collect an iterator into an array, though. One alternative way to implement this is to iterate over the slice elements and assign into some mutable array (not as pretty, but avoids the Vec)

Reply:
Hmm, why not leave the code as it was before, with a slight modification:
``` rust
let projection = Perspective3::new(aspect_ratio, fovy, znear, zfar);
let mut res = [0f64; 16];
res.copy_from_slice(
    (projection.to_projective() * self.0).matrix().as_slice(),
);
res.map(|f| Scalar::from(*f))
```

Should work?

But yeah, it's not that important. I'd be happy to merge a change that avoids the `Vec`, but I'm also fine with leaving it as-is.

Reply:
That would be a good way to avoid the Vec while still keeping things tidy. I don't think this will make or break the performance but it'll be good to keep in mind for later.

Reply:
Agreed. No reason to bend over backwards right now. If there's really a problem, we can fix it.